### PR TITLE
CAL Standard Interface (Abstract Base Class Declaration)

### LINKED ISSUE #20: Implement CAL (CANBus Abstraction Layer) Abstract Class
Write a C++ library (CAL) which will implement an abstract class for the required memory mappings of ECU/PDM signals.

### PR DESCRIPTION
**Pull request by ethanmpeterson**
Closes #20

### CODE REVIEW

File: cal/include/cal.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+};
+
+class CAL {
+    public:
+        virtual bool init() = 0;
+        virtual CALMessage receive() = 0;
+        virtual bool send(CALMessage) = 0;
+
+        // Called on message receive if the underlying hardware has
+        // Rx interrupt
```
Comment (bchampp) [inline cal/include/cal.h]: Whys this split onto a new line? It's not near the 80 character limit

File: steering/platformio.ini [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
 framework = arduino
 ; change microcontroller
 board_build.mcu = stm32f446re
 monitor_speed = 115200
 ; change MCU frequency
 board_build.f_cpu = 180000000L
-; can specify hardcoded port using upload_port flag
\ No newline at end of file
+build_flags = 
+    -I../cal/include
```
Comment (bchampp) [inline steering/platformio.ini:13]: 👍

File: steering/src/main.cpp [hunk 3]
```diff
@@ -1,5 +1,6 @@
 #include "Arduino.h"
 #include "steering_io.h"
+#include "cal.h"
```
Comment (bchampp) [inline steering/src/main.cpp:3]: Including this so that it gets added to compilation?

**Review by bchampp — APPROVED** [review]
Nice work, this is a great start to CAL 💯

File: steering/src/main.cpp (hunk 3 above)
Comment (ethanmpeterson) [reply steering/src/main.cpp:3]: yessir

**Pull request merged**